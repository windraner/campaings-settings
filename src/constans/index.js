export const MENU_LIST = [
  'Campaign details',
  'Media Collections',
  'Blacklists',
  'Media rights'
];

export const ACTIVE_TAB = 'activeTab';
export const NAME = 'name';
export const DESCRIPTION = 'description';
export const PRODUCT_CATALOGUE = 'productCatalogue';
export const IS_SHOPPABLE_LINK = 'isShoppableLink';
export const IS_MEDIA_COLLECTION = 'isMediaCollection';
export const MEDIA_COLLECTION_LIST = 'mediaCollectionList';
export const IS_PHOTO_MODERATION_QUEUE = 'isPhotosModerationQueue';
export const USER_BLACK_LIST = 'userBlackList';
export const HASHTAG_BLACK_LIST = 'hashtagBlackList';
export const ACCOUNTS_SETTINGS_ITEMS = 'accountsSettingsItems';
export const SHOW_ERROR_ALERT = 'showErrorAlert';
export const SHOW_SUCCESS_ALERT = 'showSuccessAlert';
export const IS_MEDIA_RIGHTS = 'isMediaRights';
export const HASHTAG_RIGHTS_LIST = 'hashtagRightsList';
export const TERMS_AND_CONDITIONS = 'termsAndConditions';

export const VALIDATED_FIELDS = [
  ACTIVE_TAB
];
