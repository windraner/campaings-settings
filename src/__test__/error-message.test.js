import React from 'react';
import ErrorMessage from '../components/main-button-section/alert-message/error-message';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { SHOW_ERROR_ALERT } from '../constans';

configure({ adapter: new Adapter() });

describe('<ErrorMessage />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<ErrorMessage stateHandler={function() {}} showErrorAlert={true} />);
    expect(wrapper.find('.campaigns-settings__custom-alert').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__custom-alert-close').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__custom-alert').text()).toBe('Error. Please fill out all required fields!');
  });
  it('Check for hiding', () => {
    const wrapper = shallow(<ErrorMessage stateHandler={function() {}} showErrorAlert={false} />);
    expect(wrapper.type()).toBe(null);
  });
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<ErrorMessage stateHandler={onClickFunc} showErrorAlert={true} />);
    const button = wrapper.find('.campaigns-settings__custom-alert-close');
    button.simulate('click');
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([[SHOW_ERROR_ALERT, false]]);
  });
  it('matches the snapshot show', () => {
    const tree = shallow(<ErrorMessage stateHandler={function() {}} showErrorAlert={true} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot hide', () => {
    const tree = shallow(<ErrorMessage stateHandler={function() {}} showErrorAlert={false} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
