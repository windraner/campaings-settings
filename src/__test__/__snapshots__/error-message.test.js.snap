// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> matches the snapshot hide 1`] = `""`;

exports[`<ErrorMessage /> matches the snapshot show 1`] = `
<div
  className="campaigns-settings__custom-alert campaigns-settings__custom-alert-error d-flex align-items-center justify-content-between"
>
  <span
    className="campaigns-settings__custom-alert-icon-error"
  />
  <span>
    Error. Please fill out all required fields!
  </span>
  <span
    className="campaigns-settings__custom-alert-close"
    onClick={[Function]}
  />
</div>
`;
